contrib/tree-sitter-epics-nice-print.sh

```diff
@@ -0,0 +1,129 @@
+# Black        0;30     Dark Gray     1;30
+# Red          0;31     Light Red     1;31
+# Green        0;32     Light Green   1;32
+# Brown/Orange 0;33     Yellow        1;33
+# Blue         0;34     Light Blue    1;34
+# Purple       0;35     Light Purple  1;35
+# Cyan         0;36     Light Cyan    1;36
+# Light Gray   0;37     White         1;37
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+BLUE='\033[0;34m'
+YELLOW='\033[1;33m'
+CYAN='\033[0;36m'
+NC='\033[0m'        # No Color
+# printf "I ${RED}love${NC} Stack Overflow\n"
+# echo -e "I ${RED}love${NC} Stack Overflow"
+echo -e "color code:"
+echo -e " - ${RED}error${NC}"
+echo -e " - ${YELLOW}warning${NC}"
+echo -e " - ${CYAN}information${NC}"
+echo -e " - ${GREEN}script command${NC}"
+echo -e " - ${BLUE}user input${NC}"
+
+IFS=$'\n\t'
+params="$(
+    getopt \
+        --options "hf:" \
+        --longoptions help,files: \
+        --name "$0" \
+        -- "$@"
+)"
+eval set -- "$params"
+
+printHelp=0
+files=""
+while true; do
+    case "$1" in
+    -h | --help)
+        printHelp=1
+        shift
+        ;;
+    -f | --files)
+        files="$2"
+        shift 2 # shift 2 times: for the option name and option value
+        ;;
+    *)
+        echo -e "${CYAN}[INFO]${NC} rest of the args: $1"
+        break
+        # exit 1
+        ;;
+    esac
+done
+
+if [[ "$printHelp" == 1 ]]; then
+    echo -e "${CYAN}[INFO]${NC} goal of the script: display errors of a specific type of EPICS files in a user-friendly format."
+    echo -e "${CYAN}[INFO]${NC} usage: $0 [-h|--help] [-f|--files \""'$(find <directory> -not -path <exclude> -type f -name <pattern>)'"\" : files to parse] [args...]"
+    echo -e "${CYAN}[INFO]${NC} <directory> : where to find the files, eg: ${HOME}/tops/ "
+    echo -e "${CYAN}[INFO]${NC} <exclude> : to exclude directory from find(), eg: \"./builds/\" "
+    echo -e "${CYAN}[INFO]${NC} <pattern> : find files with this pattern, eg for EPICS files: "*.cmd" or "*.template" or "*.db" or "*.sub*" or "*.st" or "*.proto" "
+    exit
+fi
+
+args_provided=$@
+echo -e "${CYAN}[INFO]${NC} arguments parsed:"
+echo -e "${CYAN}[INFO]${NC}   - printHelp: ${printHelp}"
+echo -e "${CYAN}[INFO]${NC}   - files: ${files}"
+echo -e "${CYAN}[INFO]${NC}   - other args are:" "$args_provided"
+
+# global var
+execute_command_return=0
+
+function execute_command(){
+    # $1: command description
+    # $2: command to execute
+    echo -e "${GREEN}$1${NC}: $ $2"
+    eval "$2"
+    execute_command_return=$?
+    echo -e "${GREEN}.. done${NC}"
+}
+
+execute_command "[CMD] get tree-sitter version" "tree-sitter --version"
+
+# for each file found, do:
+for file in $files; do
+    # echo -e "${CYAN}[INFO]${NC} file: $file"
+
+    # tree-sitter
+    output=$(tree-sitter parse --stat "$file" || true)
+    # echo -e "${CYAN}[INFO]${NC} output: $output"
+
+    # look for ERROR or MISSING in tree-sitter output
+    error_lines=$(echo "$output" | grep -E "ERROR|MISSING" || true)
```

Comment:
If you're not using the output, you can grep it directly. This prevents the whole output from being copied into memory each time, which can be quite big.

I tested it, and it makes the script 1.24 times faster with a set of 20 db files that I had on hand.

Adding the extra `--` also prevents people from triggering options if you have a file that starts with a `-`.

```suggestion
    # look for ERROR or MISSING in tree-sitter output
    error_lines=$(tree-sitter parse --stat -- "$file" | grep -E "ERROR|MISSING" || true)
```